pkg/kubectl/cmd/drain.go

```diff
@@ -612,19 +615,22 @@ func (o *DrainOptions) evictPods(pods []corev1.Pod, policyGroupVersion string, g
 		globalTimeout = o.Timeout
 	}
 	globalTimeoutCh := time.After(globalTimeout)
+	numPods := len(pods)
 	for {
 		select {
-		case err := <-errCh:
-			return err
-		case <-doneCh:
+		case err := <-returnCh:
 			doneCount++
-			if doneCount == len(pods) {
-				return nil
+			if err != nil {
+				errors = append(errors, err)
 			}
 		case <-globalTimeoutCh:
-			return fmt.Errorf("Drain did not complete within %v", globalTimeout)
+			return utilerrors.NewAggregate(errors)
```

Comment:
You're losing the timeout error information here, which means if it times out before the first error is reported, that's going to become a success. Maybe consider just returning timeout error in that case (as it was done before)?

Reply:
good catch on both. fixed.